### ISSUE TITLE
interpreting graphs

### ISSUE THREAD

**Issue by calvincicha** [nanoporetech/tombo#180]
I am new to tombo and looking to make some graphs that show potential base modifications. In the tutorial https://nanoporetech.github.io/tombo/tutorials.html there are some graphical outputs. link below should open them. 

https://cdn.livechat-static.com/api/file/v2/lc/att-old/7281641/f25c30a6181112ce8c30bbddae0a4211/stat_dist.png

I am having a hard time interpreting these data and how I could predict base modifications from them. I am wondering if anyone here can explain the different components of these graphs, and how to interpret them. The simpler the explanation the better.

Thanks for your time.

**Comment by marcus1487** [comment]
Since nanopore signal measures the current of a number of bases around the pore, modified bases can shift the current signal at a range of positions around a modified base. This plot is intended to help visually identify the modified base of interest within a motif by aggregating results over many genomic locations (centered on the motif of interest).

TL;DR: If there is a modified base consistently found within the provided motif in the sample provided, the bottom panel of this graph should show a peak at the modified base within the motif.

More detail:

In this plot, the top panel shows 3 example reference locations centered on the motif of interest (these are actually the very top of the statistical rank list). You can see in the ground truth modified plot you attached that signal (red squiggle-y lines) deviate from the expected signal levels given the canonical reference bases (grey bell curve distributions) at positions around the central `C` in the `CCWGG` motif (though with some dependence on the large sequence context).

The bottom panel aggregates the statistics computed at each reference position relative to the provided motif of interest over some number of reference locations (specified by `--num-statistics`; default `200`). These reference locations are obtained by iterating through the rank list of modified base statistics and filtering out locations that do not contain the motif of interest (within the range specified by the `--num-context` argument). The statistics at this set of reference locations are then aligned relative to the motif of interest and the distribution of these statistics at each relative position is plotted in the bottom panel.

Note in the second example containing a motif not modified in E. coli that the statistics show enrichment specifically outside of the provided motif. This is because the sample contains modified bases near this motif, but only in the dam and dcm (and EcoK1) contexts. Thus we see enrichment for modified bases specifically outside of the motif provided.

**Comment by calvincicha** [comment]
Thank you for this explanation. It was very helpful.

I have one more clarification.

To me it looks like the top signal graph does not agree with the bottom panel violin plot. For example, in the violin plot the second C of the C1C2WG1G2 motif appears to me the most significant, while in the top signal plot it look like the two flanking bases (C1 & T) of the actual modified base C1 are more divergent from the expected values. In my mind according to the top signal plot C1 and T should show the most significant fraction modified.

**Comment by marcus1487** [comment]
This has to do with how the statistics are computed/processed. The `tombo detect_modifications de_novo` command smooths statistical values over a moving window defined by the `--fishers-method-context` argument (default value of `1` base of context to either side resulting in a moving window of `3`) to account for the fact that shifted signal positions happen around and not necessarily at the modified base of interest. As you note, at this motif the `C1` and `T` positions are the most statistically significant, but with the moving window of 3 bases the `C2` base becomes the most significant.

In order to test for positions with the largest shift (and not necessarily the location of the modified base) set the `--fishers-method-context` argument to `0`.
